marine.c

```diff
@@ -607,14 +607,15 @@ WS_DLL_PUBLIC int validate_bpf(char *bpf) {
     return TRUE;
 }
 
-int inner_compile_dfilter(char *dfilter, dfilter_t **dfcode, char *err_msg) {
+int inner_compile_dfilter(char *dfilter, dfilter_t **dfcode, char **err_msg) {
     char *dfilter_err_msg;
 
     int compile_status = dfilter_compile(dfilter, dfcode, &dfilter_err_msg);
```

Comment:
Because this will lead to the fact that we need to `g_free` as a public function so marine-python can free the error.

Reply:
That makes me think that we shouldn't expose `free` at all, but expose `marine_free_err_msg` (or some other similar name).
It's essentially the same, but semantically different and feels less horrible (and also similar to `marine_free_result`).

Reply:
So just change the name of the function to clarify it's usage?
Should I then change it into `g_free`?  Or leave it as is to try and create the least harm?

Reply:
Yes, just rename it.
Change the invoked `free` function to whatever is appropriate.
Obvoiusly, I think that changing this line according to my suggestion is the nicer and it will require you to use `g_free`.